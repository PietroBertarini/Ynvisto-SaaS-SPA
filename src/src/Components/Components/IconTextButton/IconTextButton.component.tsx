import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { Link } from 'react-router-dom';
import React, { useCallback } from 'react';
import { ListItem, SvgIcon, Link as hrefLink } from '@material-ui/core';
import Swal from 'sweetalert2';
import { useStyles, EmBreve } from './IconTextButton.styles';

export interface OwnProps {
    /**
     * Showing text of button
     */
    label: string;
    /**
     * Image Icon path of button
     */
    iconPath: any;
    /**
     * Link Path of Ynvisto or text to be shown if functionality not released yet
     */
    path: string;
    /**
     * external Link Path of Ynvisto
     */
    href?: boolean;
    /**
     * functionality not released yet
     */
    commingSoon?: boolean;
    /**
     * active label
     */
    active?: boolean;
    /**
     * onClick function
     */
    onClick?: () => void;
}

/**
 * Component from @LeftMenu that have a icon and text Button, when clicked
 * goes to a @path link.
 * @param label: The showing text from button
 * @param iconPath: IconPath to render at the left of @label.
 * @param path: The url link that the Ynvisto goes when the button is clicked.
 */
function IconTextButton({
  label, iconPath, path, onClick, href, commingSoon, active,
} : OwnProps) {
  const classes = useStyles();
  const showCommingSoon = useCallback((text: string) => {
    Swal.fire({
      icon: 'info',
      title: text,
    });
  }, []);

  if (href) {
    return (
      <ListItem onClick={onClick} button component={hrefLink} target="_blank" href={path} className={classes.listItem}>
        <ListItemIcon style={{ minWidth: 40 }}>
          <SvgIcon style={{ color: '#B1B1B1' }} component={iconPath} viewBox="0 0 600 476.6" />
        </ListItemIcon>
        <ListItemText primary={label} style={{ color: '#B1B1B1' }} />
      </ListItem>
    );
  }

  if (commingSoon) {
    return (
      <ListItem button onClick={() => showCommingSoon(path)} className={classes.listItem}>
        <ListItemIcon style={{ minWidth: 40 }}>
          <SvgIcon style={{ color: '#B1B1B1' }} component={iconPath} viewBox="0 0 600 476.6" />
        </ListItemIcon>
        <ListItemText primary={label} style={{ color: '#B1B1B1' }} />
        <EmBreve>em breve</EmBreve>
      </ListItem>
    );
  }

  if (active) {
    return (
      <ListItem onClick={onClick} button className={classes.active}>
        <ListItemIcon style={{ minWidth: 40 }}>
          <SvgIcon style={{ color: '#fff' }} component={iconPath} viewBox="0 0 600 476.6" />
        </ListItemIcon>
        <ListItemText primary={label} style={{ color: '#fff' }} />
      </ListItem>
    );
  }

  return (
    <ListItem onClick={onClick} button component={Link} to={path} className={classes.listItem}>
      <ListItemIcon style={{ minWidth: 40 }}>
        <SvgIcon style={{ color: '#B1B1B1' }} component={iconPath} viewBox="0 0 600 476.6" />
      </ListItemIcon>
      <ListItemText primary={label} style={{ color: '#B1B1B1' }} />
    </ListItem>
  );
}
export default IconTextButton;
