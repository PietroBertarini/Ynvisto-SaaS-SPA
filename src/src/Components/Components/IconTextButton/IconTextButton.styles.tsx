import { createStyles, makeStyles } from '@material-ui/core/styles';
import styled from 'styled-components';

export const useStyles = makeStyles(() => createStyles({
  listItem: {
    '& .Mui-selected,&:hover': {
      backgroundColor: 'rgba(255, 255, 255, 0.16)',
      '&:hover': { backgroundColor: 'rgba(255, 255, 255, 0.16)' },
    },
    marginTop: 7,
  },
  active: {
    borderLeft: '3px solid #E9C46A',
    borderRadius: 3,
    color: '#B1B1B1',
    marginTop: 7,
    '&:hover': { backgroundColor: 'rgba(255, 255, 255, 0.16)' },
  },
}));

export const EmBreve = styled.div`
  border: solid 1px #E9C46A;
  border-radius: 10px;
  color: #E9C46A;
  padding: 0 3px;
`;
