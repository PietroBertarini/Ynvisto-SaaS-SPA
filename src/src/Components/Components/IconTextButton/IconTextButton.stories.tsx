import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import IconTextButton, { OwnProps } from './IconTextButton.component';
import { ReactComponent as acoesIcon } from '../../../Assets/Images/trandingIcon.svg';

export default {
  title: 'Components/LeftMenu/SubComponents/IconTextButton',
  component: IconTextButton,
} as Meta;

const Template: Story<OwnProps> = (args) => <IconTextButton {...args} />;

export const Default = Template.bind({});
Default.args = {
  path: '/Acoes',
  iconPath: acoesIcon,
  label: 'Acoes',
};
