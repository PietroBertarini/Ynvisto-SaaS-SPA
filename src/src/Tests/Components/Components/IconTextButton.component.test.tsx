import React from 'react';
import { shallow } from 'enzyme';
import IconTextButton from '../../../Components/Components/IconTextButton/IconTextButton.component';
import { ReactComponent as whiteWallet } from '../../../Assets/Images/WalletIcon/whiteWallet.svg';

test('Renders IconTextButton component and match to previously snapshot', () => {
  const mockProps = {
    label: 'Resumo',
    svgIconPath: whiteWallet,
    path: '/resumo',
  };

  const component = shallow(
    <IconTextButton
      label={mockProps.label}
      iconPath={mockProps.svgIconPath}
      path={mockProps.path}
    />,
  );
  expect(component).toMatchSnapshot();
});
