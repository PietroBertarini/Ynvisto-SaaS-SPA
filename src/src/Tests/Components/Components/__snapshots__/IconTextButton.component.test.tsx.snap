// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders IconTextButton component and match to previously snapshot 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  className="makeStyles-listItem-1"
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "Link",
      "propTypes": Object {
        "innerRef": [Function],
        "onClick": [Function],
        "replace": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  to="/resumo"
>
  <WithStyles(ForwardRef(ListItemIcon))
    style={
      Object {
        "minWidth": 40,
      }
    }
  >
    <WithStyles(ForwardRef(SvgIcon))
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      style={
        Object {
          "color": "#B1B1B1",
        }
      }
      viewBox="0 0 600 476.6"
    />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    primary="Resumo"
    style={
      Object {
        "color": "#B1B1B1",
      }
    }
  />
</WithStyles(ForwardRef(ListItem))>
`;
